cpp/src/arrow/compute/kernels/test_util.cc

```diff
@@ -263,6 +263,23 @@ void CheckScalarUnary(std::string func_name, std::shared_ptr<DataType> in_ty,
 
 void CheckVectorUnary(std::string func_name, Datum input, Datum expected,
                       const FunctionOptions* options) {
+  // WTF debug
+  std::cout << "input: ";
+  switch (input.kind()) {
+    case Datum::ARRAY:
+      std::cout << input.make_array()->ToString();
+      break;
+    case Datum::CHUNKED_ARRAY:
+      std::cout << input.chunked_array()->ToString();
+      break;
+    case Datum::SCALAR:
+      std::cout << input.scalar()->ToString();
+      break;
+    default:
+      std::cout << input.ToString();
+      break;
+  }
+  std::cout << std::endl;
```

Comment:
Should this have been merged?

Reply:
Woops. Definitely not. Sorry, I should have been more careful here. @bkietz Do you want to prepare a fixup PR?

Reply:
It seems to have been removed and I was confused by the GitHub notification showing me this code as green.

Reply:
This is not on main: https://github.com/apache/arrow/blob/96ac514383bc33b02d19d739a6428ad9b3f453b7/cpp/src/arrow/array/data.cc#L266

https://github.com/apache/arrow/commit/96ac514383bc33b02d19d739a6428ad9b3f453b7#diff-1d848527e898cd9da6a41bdbd68cab5d5826986128851220d20d1fdac9752d8b